metricbeat/module/kubernetes/state_pod/_meta/data.json

```diff
@@ -6,17 +6,22 @@
         "module": "kubernetes"
     },
     "kubernetes": {
-        "namespace": "default",
+        "namespace": "kube-system",
         "node": {
             "name": "kind-control-plane"
         },
         "pod": {
             "host_ip": "172.18.0.2",
-            "ip": "10.244.0.22",
-            "name": "hello-28192809-jbn5z",
+            "ip": "10.244.0.3",
+            "name": "coredns-787d4945fb-kx4f4",
+            "owner": {
+                "is_controller": "true",
+                "kind": "ReplicaSet",
+                "name": "coredns-787d4945fb"
```

Comment:
It does not add the deployment name correct only 1st level, you are right.

I was thinking that we can justify that this is enough for the users in order to figure out the reference. Of course we need to document this. Otherwise even if we can try in code or with ingest pipelines to trim the strings and produce ancestors we might fall in mistakes because we are not sure for the existance.

Also if we integrate this https://github.com/elastic/elastic-agent-autodiscover/pull/62, this means that also the enrichers will loose those metadata is not it?

Reply:
Through the Enricher every metrics' document for Pod's like `pod.cpu.usage` will be enriched with Pod's metadata. In this metadata the Enricher adds the owner name like `kubernetes.cronjob.name`, `kubernetes.replicaset.name` etc. So this is not related to the settings you change at https://github.com/elastic/elastic-agent-autodiscover/pull/62. That's a diffrent feature takes place at https://github.com/elastic/elastic-agent-autodiscover/blob/2cc3dcb075fe437467387d36096f554c13c6d144/kubernetes/metadata/pod.go#L96-L122.

So still this patch seems to be redundant. 
Could you try to run Metricbeat with k8s module enabled and see that Pod's metrics are enriched with the owner name (disable the deployment and cronjob metadata as usually to isolate the functionality)? This will help you understand how the feature works and that this patch actually overlaps already existent information.